### DISCUSSION TITLE
Is it possible to set the log format for init and wait containers to JSON?

### DISCUSSION THREAD

**Discussion by argl** [argoproj/argo-workflows#7361] [Q&A] [answered]
I have looked at the code, and it seems that this is currently not possible. The arguments passed to argoexec are hardcoded:
```
func (woc *wfOperationCtx) newInitContainer(tmpl *wfv1.Template) apiv1.Container {
	ctr := woc.newExecContainer(common.InitContainerName, tmpl)
	ctr.Command = []string{"argoexec", "init", "--loglevel", getExecutorLogLevel()}
	return *ctr
}
```

Am I correct? Ideally I would expect that the configured log format (text or json) in the workflow controller is passed down to the init/wait container declarations it creates. Would that be a useful ehancement?

**Comment by Hunter-Thompson** [comment, marked as answer]
#9169 should fix this.
